Python/pywarpx/_libwarpx.py

```diff
@@ -30,112 +30,294 @@
+        self.libwarpx.warpx_getCellSize.argtypes = [ctypes.c_int, ctypes.c_int]
+        self.libwarpx.warpx_getistep.argtypes = [ctypes.c_int]
+        self.libwarpx.warpx_setistep.argtypes = [ctypes.c_int, ctypes.c_int]
+        self.libwarpx.warpx_gett_new.argtypes = [ctypes.c_int]
+        self.libwarpx.warpx_sett_new.argtypes = [ctypes.c_int, c_real]
+        self.libwarpx.warpx_getdt.argtypes = [ctypes.c_int]
+
+    def get_boundary_number(self, boundary):
+        '''
+
+        Utility function to find the boundary number given a boundary name.
+
+        Parameters
+        ----------
+
+            boundary       : the boundary from which to get the scraped particle data.
+                             In the form x/y/z_hi/lo or eb.
+
+        Returns
+        -------
+
+        Integer index in the boundary scraper buffer for the given boundary.
+        '''
+        if self.geometry_dim == '3d':
+            dimensions = {'x' : 0, 'y' : 1, 'z' : 2}
+        elif self.geometry_dim == '2d':
+            dimensions = {'x' : 0, 'z' : 1}
+        elif self.geometry_dim == '1d':
+            dimensions = {'z' : 0}
+        else:
+            raise NotImplementedError("RZ is not supported for particle scraping.")
+
+        if boundary != 'eb':
+            boundary_parts = boundary.split("_")
+            dim_num = dimensions[boundary_parts[0]]
+            if boundary_parts[1] == 'lo':
+                side = 0
+            elif boundary_parts[1] == 'hi':
+                side = 1
+            else:
+                raise RuntimeError(f'Unknown boundary specified: {boundary}')
+            boundary_num = 2 * dim_num + side
+        else:
+            if self.geometry_dim == '3d':
+                boundary_num = 6
+            elif self.geometry_dim == '2d':
+                boundary_num = 4
+            elif self.geometry_dim == '1d':
+                boundary_num = 2
 
-# our particle data type, depends on _ParticleReal_size
-_p_struct = [(d, _numpy_particlereal_dtype) for d in 'xyz'[:dim]] + [('id', 'i4'), ('cpu', 'i4')]
-_p_dtype = np.dtype(_p_struct, align=True)
+        return boundary_num
 
-_numpy_to_ctypes = {}
-_numpy_to_ctypes[_numpy_particlereal_dtype] = c_particlereal
-_numpy_to_ctypes['i4'] = ctypes.c_int
 
-class Particle(ctypes.Structure):
-    _fields_ = [(v[0], _numpy_to_ctypes[v[1]]) for v in _p_struct]
 
+libwarpx = LibWarpX()
 
-# some useful typenames
-_LP_particle_p = ctypes.POINTER(ctypes.POINTER(Particle))
-_LP_c_int = ctypes.POINTER(ctypes.c_int)
-_LP_LP_c_int = ctypes.POINTER(_LP_c_int)
-_LP_c_void_p = ctypes.POINTER(ctypes.c_void_p)
-_LP_c_real = ctypes.POINTER(c_real)
-_LP_LP_c_real = ctypes.POINTER(_LP_c_real)
-_LP_c_particlereal = ctypes.POINTER(c_particlereal)
-_LP_LP_c_particlereal = ctypes.POINTER(_LP_c_particlereal)
-_LP_c_char = ctypes.POINTER(ctypes.c_char)
-_LP_LP_c_char = ctypes.POINTER(_LP_c_char)
+# --- Is there a better way of handling constants?
+clight = 2.99792458e+8 # m/s
```

Comment:
I believe that constants are stored in the `constants` class of `picmi.py`. `clight` could be moved there to keep all the constants in one place.